### PR TITLE
HM_hibernate06

### PR DESCRIPTION
**Pull request by grom4er**

### CODE REVIEW

File: src/main/java/cinema/Application.java [hunk 1]
```diff
@@ -54,7 +58,27 @@ public static void testUserAndAuthentication()
         user.setEmail("tutut@gmal.com");
         user.setPassword("123");
         authenticationService.register(user.getEmail(), user.getPassword());
-        authenticationService.login(user.getEmail(), user.getPassword());
-        //  authenticationService.login(user.getEmail(),"1222");
+        System.out.println(authenticationService.login(user.getEmail(), user.getPassword()));
+        // authenticationService.login(user.getEmail(),"1222");
```
Comment (Max-Pochepets) [inline src/main/java/cinema/Application.java:62]: delete this please

File: src/main/java/cinema/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapsId;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shoppingCart")
```
Comment (Max-Pochepets) [inline src/main/java/cinema/model/ShoppingCart.java:14]:
```suggestion
@Table(name = "shopping_carts")
```
same for others

File: src/main/java/cinema/service/ShoppingCartService.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import cinema.model.ShoppingCart;
+import cinema.model.User;
+
+public interface ShoppingCartService {
+    /**
+     * This method is responsible for adding a Ticket to the ShoppingCart
+     *
+     * @param movieSession contains the information required for the ticket
+     * @param user         - the User who wants to buy the ticket for a specific movieSession
+     */
```
Comment (Max-Pochepets) [inline src/main/java/cinema/service/ShoppingCartService.java:13]: #remove

File: src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ShoppingCart shoppingCart = getByUser(user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
+
```
Comment (Max-Pochepets) [inline src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java:28]:
```suggestion
```

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/cinema/Application.java [hunk 5]
```diff
@@ -54,7 +58,26 @@ public static void testUserAndAuthentication()
...
         authenticationService.register(user.getEmail(), user.getPassword());
-        authenticationService.login(user.getEmail(), user.getPassword());
-        //  authenticationService.login(user.getEmail(),"1222");
+        System.out.println(authenticationService.login(user.getEmail(), user.getPassword()));
+    }
+
+    public static void testCartService() {
+        User user = new User();
+        user.setEmail("adriano@gmail.com");
+        user.setPassword("123");
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/Application.java:67]: let's save this user to the db

File: src/main/java/cinema/model/MovieSession.java [hunk 6]
```diff
@@ -16,12 +14,10 @@
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    @ManyToOne(fetch = FetchType.LAZY)
-    @JoinColumn(name = "movie_id")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/MovieSession.java:20]: there is no need to remove `@JoinColumn` annotations)

File: src/main/java/cinema/model/ShoppingCart.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapsId;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_carts")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/ShoppingCart.java:14]: let's use either singular or plural table names for all the tables

File: src/main/java/cinema/model/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @OneToMany(cascade = CascadeType.ALL)
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/ShoppingCart.java:18]: let's remove this - we should be very careful with cascades - especially with those that include DELETE operation

File: src/main/java/cinema/model/Ticket.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "tickets")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "movieId")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/model/Ticket.java:18]:
```suggestion
    @JoinColumn(name = "movie_id")
```
remember about the correct column naming

File: src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import cinema.model.MovieSession;
+import cinema.model.ShoppingCart;
+import cinema.model.Ticket;
+import cinema.model.User;
+import cinema.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private TicketDao ticketDao;
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java:16]: let's use this to save the created ticket to the DB - but do this after you already get the shoppingCart from the DB

File: src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public void registerNewShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public void clear(ShoppingCart shoppingCart) {
+        ShoppingCart shoppingCartByUser = getByUser(shoppingCart.getUser());
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java:44]: do you need this?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**